inc/inventory/asset/software.class.php

```diff
@@ -104,14 +106,20 @@ public function prepare() :array {
             }
          }
 
-         if (!property_exists($val, 'name') || (property_exists($val, 'name') && $val->name == '')) {
+         if (!property_exists($val, 'name')
+            || ($val->name == ''
+               || Toolbox::startsWith(Toolbox::slugify($val->name), 'nok_')
+            )
+         ) {
             if (property_exists($val, 'guid') && $val->guid != '') {
                $val->name = $val->guid;
             }
          }
 
          //If the software name exists and is defined
          if (property_exists($val, 'name') && $val->name != '') {
+            $val->name = trim(str_replace("\xc2\xa0", ' ', $val->name));
```

Comment:
Unless I am wrong, `\xc2\xa0` is part of whitespace chars in PCRE regex. Maybe it would be more efficient to directly handle all uncommon spacing chars.

```suggestion
            $val->name = trim(preg_replace('/\s+/', ' ', $val->name));
```

Reply:
Well, I dod not see the case, and a regexp is often less performant than a str_replace